### PR TITLE
Deprecate the 'sameFile' overload with 'file' arguments

### PR DESCRIPTION
**Pull request by daviditen**
The 'sameFile' function that takes 'file' arguments is the only function in the 'FileSystem' module that takes 'file' arguments. Deprecate it for the sake of consistency with the rest of the functions in 'FileSystem'.

Update tests to use the 'string' version of 'sameFile' where 'file.path' is available. Stop testing on a few tests that were calling 'sameFile' on an unset 'file'.

Add a deprecated test for 'sameFile'.

### CODE REVIEW

**Comment by daviditen** [comment]
Passed Linux64 paratest.

File: test/library/standard/FileSystem/lydia/sameFile/givenNull.notest [hunk 1]
```diff
@@ -0,0 +1,3 @@
+This is testing 'sameFile' on a default initialized 'file' record.
```
Comment (lydia-duncan) [inline test/library/standard/FileSystem/lydia/sameFile/givenNull.notest:1]: Why not just delete these tests?

**Review by lydia-duncan — APPROVED** [review]
Nice!  I think you can just remove the NULL tests instead of marking them as `notest`.  If we want sameFile's file argument functionality back, reverting the PR or looking at this PR will reveal the tests were deleted

**Comment by daviditen** [comment]
Thanks! I removed the NULL tests.

**Pull request merged**